packages/lix-sdk/src/observe/create-observe.ts

```diff
@@ -28,6 +28,170 @@ function areResultsEqual<T>(a: T[], b: T[]): boolean {
 	return JSON.stringify(a) === JSON.stringify(b);
 }
 
+/**
+ * Maps table names to their corresponding schema keys.
+ * This is the reverse mapping of what's done in schema.ts.
+ */
+export function determineSchemaKeys(compiledQuery: any): string[] {
+	const tableNames = new Set<string>();
+
+	// Extract table names from the compiled query
+	try {
+		// The compiled query has a 'query' property that contains the actual AST
+		const queryNode = compiledQuery.query || compiledQuery;
+
+		// Get table names from FROM clause
+		if (queryNode.from) {
+			extractTableNamesFromQueryNode(queryNode.from, tableNames);
+		}
+
+		// Get table names from JOIN clauses
+		if (queryNode.joins) {
+			for (const join of queryNode.joins) {
+				extractTableNamesFromQueryNode(join, tableNames);
+			}
+		}
+
+		// Get table names from WHERE clause subqueries
+		if (queryNode.where) {
+			extractTableNamesFromQueryNode(queryNode.where, tableNames);
+		}
+
+		// Get table names from SELECT clause subqueries
+		if (queryNode.selections) {
+			for (const selection of queryNode.selections) {
+				extractTableNamesFromQueryNode(selection, tableNames);
+			}
+		}
+	} catch (error) {
+		console.warn("Failed to extract table names from compiled query:", error);
+		return []; // Return empty array to fall back to always re-executing
+	}
+
+	// Map table names to schema keys
+	const tableToSchemaMap: Record<string, string> = {
+		key_value: "lix_key_value",
+		version: "lix_version",
+		change_set: "lix_change_set",
+		file: "lix_file",
+		account: "lix_account",
+		label: "lix_label",
+		thread: "lix_thread",
+		thread_comment: "lix_thread_comment",
+		change_author: "lix_change_author",
+		log: "lix_log",
+		stored_schema: "lix_stored_schema",
+		change_set_element: "lix_change_set_element",
+		change_set_edge: "lix_change_set_edge",
+		change_set_label: "lix_change_set_label",
+		change_set_thread: "lix_change_set_thread",
+		// Add special tables that don't have direct schema mappings but are important
+		change: "change", // Special case for change table
+		state: "state", // Virtual state table - could include multiple schema keys
+		active_version: "lix_version", // Maps to version schema
+		active_account: "lix_account", // Maps to account schema
+	};
```

Comment:
Import the `LixSchemaViewMap` from `src/database/schema.ts` for a non hardcoded version. if you get a circular dependency issue, move the schema view map into its own file in `src/database/schema-view-map.ts`